stylelint.config.js

```diff
@@ -0,0 +1,225 @@
+
+    'function-calc-no-invalid': [true, { severity: 'warning' }],
+    'function-calc-no-unspaced-operator': [true, { severity: 'warning' }],
+    'function-linear-gradient-no-nonstandard-direction': [true, { severity: 'warning' }],
+
+    'string-no-newline': [true, { severity: 'warning' }],
+
+    'unit-no-unknown': [true, { severity: 'warning' }],
+
+    'property-no-unknown': [true, { severity: 'warning' }],
+
+    'keyframe-declaration-no-important': [true, { severity: 'warning' }],
+
+    'declaration-block-no-duplicate-properties': [true, { severity: 'warning' }],
+    'declaration-block-no-shorthand-property-overrides': [true, { severity: 'warning' }],
+
+    'block-no-empty': [true, { severity: 'warning' }],
+
+    'selector-pseudo-class-no-unknown': [true, { severity: 'warning' }],
+    'selector-pseudo-element-no-unknown': [true, { severity: 'warning' }],
+    'selector-type-no-unknown': [true, { severity: 'warning' }],
+
+    'media-feature-name-no-unknown': [true, { severity: 'warning' }],
+
+    'at-rule-no-unknown': [true, {
+      severity: 'warning',
+      ignoreAtRules: ['mixin', 'include'],
+    }],
+
+    'comment-no-empty': [true, { severity: 'warning' }],
+
+    'no-duplicate-at-import-rules': [true, { severity: 'warning' }],
+    'no-duplicate-selectors': [true, { severity: 'warning' }],
+    'no-extra-semicolons': [true, { severity: 'warning' }],
+
+    'number-max-precision': [6, { severity: 'warning' }],
+
+    'time-min-milliseconds': [100, { severity: 'warning' }],
+
+    'unit-blacklist': [['px', 'cm', 'mm', 'pt', 'in', 'pc'], {
+      severity: 'warning',
+      /**
+       * it's not working with '["px", "dpi"]' syntax as in example in the docs
+       * so we need to use this hack with regex.
+       */
+      ignoreMediaFeatureNames: {
+        'px': [/a-zA-Z0-9/],
+        'dpi': [/a-zA-Z0-9/],
+      },
+      /**
+       * allow to use 'px' and 'dpi' in variables that has prefix 'media-'
+       *
+       * we need to use "." (dot) at the beginning of the regex to prevent
+       * handle variables values because linter doesn't read '$' value in the
+       * "/^\$media-/" regex, so we try not to match the first character
+       */
+      ignoreProperties: {
+        'px': ['/^.media-/'],
+        'dpi': ['/^.media-/'],
+      },
+    }],
+
+    'shorthand-property-no-redundant-values': [true, { severity: 'warning' }],
+
+    'value-no-vendor-prefix': [true, { severity: 'warning' }],
+
+    'property-no-vendor-prefix': [true, { severity: 'warning' }],
+
+    'declaration-block-no-redundant-longhand-properties': [true, { severity: 'warning' }],
+    'declaration-no-important': [true, { severity: 'warning' }],
+
+    'declaration-block-single-line-max-declarations': [1, { severity: 'warning' }],
+
+    'selector-max-empty-lines': [0, { severity: 'warning' }],
+
+    'at-rule-no-vendor-prefix': [true, { severity: 'warning' }],
+
+    'no-unknown-animations': [true, { severity: 'warning' }],
```

Comment:
OK let’s just increase the restrain to 3 to cover these cases.

Reply:
> if stylelint can provide us with other solution, like "do not nest anything but pseudselectors" it would be nice
[selector-nested-pattern](https://stylelint.io/user-guide/rules/selector-nested-pattern/#selector-nested-pattern) rule looks like what we need

Reply:
yep, nice thing, thanks. for the cases when we do need nesting (markdown-parser or other 3rd-party lib) we can do `/* stylelint-disable selector-nested-pattern */`